Assets/Scripts/VoiceCommandController.cs

```diff
@@ -141,23 +143,35 @@ private void FaceDestination()
         transform.rotation = Quaternion.Slerp(transform.rotation, lookRotation, Time.deltaTime * 10f);
     }
 
-    public void DialogueVariableHandler(string command)
+    public void DialogueVariableHandler(string input)
     {
-        command = command.Trim().ToLower();
-
-        DialogueLua.SetVariable("Input", command);
-        //!!!!!!!IMPORTANT!!!!!!! ALL INPUT IS READ LOWERCASE YOU STUPID FUCK!!!!!!!! DO NOT PUT ANY CAPS IN THE "CONTAINS" STATEMENT!!!!!!!!!!!!!!!!! Okay thanks :D 
-        if (command.Contains("yes") || command.Contains("yeah") || command.Contains("yep") || command.Contains("i do") || command.Contains("i can") || command.Contains("i'm here"))
-        {
-            DialogueLua.SetVariable("Y/N/E", 1);
-        }
-        else if(command.Contains("no") || command.Contains("nah") || command.Contains("nope"))
+        // Check if the current scene is "IntroScene"
+        if (SceneManager.GetActiveScene().name == "IntroScene")
         {
-            DialogueLua.SetVariable("Y/N/E", 2);
+            var firstResponse = DialogueManager.currentConversationState.pcResponses[0];
+            input = input.Trim().ToLower();
+            DialogueLua.SetVariable("Input", input);
+
+            // All input is read lowercase
+            if (input.Contains("yes") || input.Contains("yeah") || input.Contains("yep") || input.Contains("i do") || input.Contains("i can") || input.Contains("i'm here"))
```

Comment:
or maybe in case they aren't all yes or no questions, you could support maybe three arrays of answers, one for each possible branch in your dialogue conversation tree (this would limit it to 3 branches, which seems reasonable, but you could pick a higher number).

So it would be like
```csharp
int DetermineConversationBranch(string playerInput, string[] branch1Answers, string[] branch2Answers = null, string branch3Answers = null)
{
   if (branch1Answers.Any(playerInput.Contains))
      return 1;

// I think ?. can automatically check for nulls but not sure if Unity supports it here. If not, you'd need a separate null check to see if we passed in any branch2Answers or branch3Answers
   if (branch2Answers?.Any(playerInput.Contains))
      return 2;

   if (branch3Answers?.Any(playerInput.Contains))
      return 3;

// maybe 0 is for cases where we don't recognize any of the input. Or you could do -1. Doesn't matter.
   return 0;
}
``` 
Giving defaults for the branch2 and branch3 as null means you don't have to provide those when you call the function if you don't need them.

Reply:
so using it would be like:

``` csharp
// this could be a constant that is reused a lot!
// imagine something similar for yes.
string[] NoAnswers = new string[] {"no", "na", "nope"};
var responseIndex = GetResponse(playerInput, yesAnswers, noAnswers);
var response = DialogueManager.currentConversationState.pcResponses[responseIndex];
DialogueManager.standardDialogueUI.OnClick(response);
```